lib/modules/datasource/maven/util.ts

```diff
@@ -91,13 +93,78 @@ export async function downloadHttpProtocol(
       // istanbul ignore next
       logger.debug({ failedUrl }, 'Unsupported host');
     } else {
-      logger.info({ failedUrl, err }, 'Unknown error');
+      logger.info({ failedUrl, err }, 'Unknown HTTP download error');
     }
     return {};
   }
 }
 
-export async function checkHttpResource(
+function isS3CedentialsError(err: { name: string; message: string }): boolean {
+  return err.name === 'CredentialsProviderError';
+}
+
+function isS3RegionError(err: { name: string; message: string }): boolean {
+  return err.message === 'Region is missing';
+}
+
+function isS3NotFound(err: { name: string; message: string }): boolean {
+  return err.message === 'NotFound' || err.message === 'NoSuchKey';
+}
+
+function parseS3Url(rawUrl: url.URL | string): { Bucket: string; Key: string } {
+  const parsedUrl = (
+    typeof rawUrl === 'string' ? url.parse(rawUrl, false) : rawUrl
```

Comment:
use our `parseUrl` from `util/url`

Reply:
Done

Reply:
please read https://github.com/renovatebot/renovate/blob/main/.github/contributing.md#resolve-review-comments-instead-of-commenting